src/domain/gift-cards/manage/index.tsx

```diff
@@ -0,0 +1,241 @@
+  }, [giftCard, isSuccess, reset])
+
+  useEffect(() => {
+    if (giftCard?.type) {
+      setType({ value: giftCard.type.value, label: giftCard.type.value })
+    }
+  }, [giftCard])
+
+  const StatusComponent = () => {
+    switch (giftCard?.status) {
+      case "published":
+        return <StatusDot title="Published" variant="success" />
+      case "draft":
+        return <StatusDot title="Draft" variant="default" />
+      case "proposed":
+        return <StatusDot title="Proposed" variant="warning" />
+      case "rejected":
+        return <StatusDot title="Rejected" variant="danger" />
+      default:
+        return null
+    }
+  }
+
+  return (
+    <div>
+      <BreadCrumb
+        currentPage={"Manage Gift Card"}
+        previousBreadcrumb={"Gift Cards"}
+        previousRoute="/a/gift-cards"
+      />
+      <form className="flex flex-col space-y-4">
+        <BodyCard
+          title="Product information"
+          subtitle="Manage the settings for your Gift Card"
+          className={"h-auto w-full"}
+          status={<StatusComponent />}
+          actionables={[
+            {
+              label:
+                giftCard?.status !== "published"
+                  ? "Publish Gift Card"
+                  : "Unpublish Gift Card",
+              onClick: () => {
+                if (giftCard?.status === "published") {
+                  submit({ status: "draft" })
+                } else {
+                  submit({ status: "published" })
+                }
+              },
+              icon: <UnpublishIcon size="16" />,
+            },
+            {
+              label: "Delete Gift Card",
+              onClick: () => deleteGiftCard(),
+              variant: "danger",
+              icon: <TrashIcon size="16" />,
+            },
+          ]}
+          events={[
+            {
+              label: "Save",
+              type: "button",
+              onClick: handleSubmit(submit),
+            },
+          ]}
+        >
+          <div className="flex flex-col space-y-6">
+            <div className="flex space-x-8">
+              <div className="flex flex-col w-1/2 space-y-4">
+                <Input
+                  label="Name"
+                  name="title"
+                  defaultValue={giftCard?.title}
+                  ref={register}
+                />
+                <Input
+                  label="Subtitle"
+                  name="subtitle"
+                  placeholder="Add a subtitle"
+                  defaultValue={giftCard?.subtitle}
+                  ref={register}
+                />
+              </div>
+              <Input
+                label="Description"
+                name="description"
+                placeholder="Add a description"
+                defaultValue={giftCard?.description}
+                className="w-1/2"
+                ref={register}
+              />
+            </div>
+            <DetailsCollapsible triggerProps={{ className: "ml-2" }}>
```

Comment:
we should probably add a padding bottom to this (just while we use save buttons)

![image](https://user-images.githubusercontent.com/88927411/151156635-126cb566-cd1f-4a05-aa16-7d0f7a5de982.png)

![image](https://user-images.githubusercontent.com/88927411/151156722-2be2fea9-3ec7-4751-9234-5a88aae5f707.png)